src/org/zaproxy/zap/extension/codedx/security/CompositeX509TrustManager.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.codedx.security;
+
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * Represents an ordered list of {@link X509TrustManager}s with additive trust.
+ * If any one of the composed managers trusts a certificate chain, then it is
+ * trusted by the composite manager.
+ * 
+ * This is necessary because of the fine-print on {@link SSLContext#init}: Only
+ * the first instance of a particular key and/or trust manager implementation
+ * type in the array is used. (For example, only the first
+ * javax.net.ssl.X509KeyManager in the array will be used.)
+ * 
+ * <a href=
+ * "http://stackoverflow.com/questions/1793979/registering-multiple-keystores-in-jvm"
+ * >see StackOverflow</a>
```

Comment:
Might be better to link here: http://codyaray.com/2013/04/java-ssl-with-multiple-keystores (or both even...)